app/src/test/java/net/squanchy/schedule/FirestoreScheduleServiceTest.kt

```diff
@@ -1,136 +1,341 @@
+                        startTime = A_START_TIME.plusOneDay().toLocalDateTime(),
+                        endTime = AN_END_TIME.plusOneDay().toLocalDateTime()
+                    ),
+                    anEvent(
+                        id = "5",
+                        numericId = 5,
+                        startTime = A_START_TIME.plusOneDay().toLocalDateTime(),
+                        endTime = AN_END_TIME.plusOneDay().toLocalDateTime()
+                    ),
+                    anEvent(
+                        id = "6",
+                        numericId = 6,
+                        startTime = A_LATER_START_TIME.plusOneDay().toLocalDateTime(),
+                        endTime = A_LATER_END_TIME.plusOneDay().toLocalDateTime()
+                    )
+                ))
+        )))
     }
 
     @Test
-    fun `verify unselected tracks are excluded from the schedule`() {
-        val schedule = aFirestoreSchedulePage(
+    fun `should not exclude any events from the schedule when not filtering only by favorites`() {
+        val schedulePage = aFirestoreSchedulePage(
             events = listOf(
-                aFirestoreEvent(track = aFirestoreTrack(id = "ABC")),
-                aFirestoreEvent(track = aFirestoreTrack()),
-                aFirestoreEvent(track = aFirestoreTrack()),
-                aFirestoreEvent(track = aFirestoreTrack(id = "BDE")),
-                aFirestoreEvent(track = aFirestoreTrack(id = "ABC"))
+                aFirestoreEvent(id = "1", track = null),
+                aFirestoreEvent(id = "2", track = null)
             )
         )
+        `when`(dbService.scheduleView()).thenReturn(Observable.just(listOf(schedulePage)))
+        val allowedTracks = setOf(aTrack(id = "a track id"))
+        `when`(tracksFilter.selectedTracks).thenReturn(BehaviorSubject.createDefault(allowedTracks))
+        `when`(checksum.getChecksumOf("1")).thenReturn(1)
+        `when`(checksum.getChecksumOf("2")).thenReturn(2)
+        // TODO mark one of the two firestore events above as favorite
+        val subscription = TestObserver<Schedule>()
 
-        `when`(dbService.scheduleView()).thenReturn(Observable.just(listOf(schedule)))
-        `when`(filterScheduleRepository.filters).thenReturn(BehaviorSubject.createDefault(setOf(aFirestoreTrack(id = "ABC").toTrack())))
+        scheduleService.schedule(onlyFavorites = false)
+            .subscribe(subscription)
 
-        val result = scheduleService.schedule(false)
-            .test()
-            .values()
-            .first()
+        subscription.assertValue(
+            aSchedule(listOf(
+                aSchedulePage(events = listOf(
+                    anEvent(id = "1", numericId = 1, track = Optional.absent()),
+                    anEvent(id = "2", numericId = 2, track = Optional.absent())
+                ))
+            ))
+        )
+    }
+
+    // TODO test events are filtered out when onyFavorites and they're not favorited
```

Comment:
We miss a test case for favourites, which depends on having favourites to make sense.